starter-code/src/functions-and-arrays.js

```diff
@@ -10,16 +18,51 @@ var words = [
   'orchard',
   'crackpot'
 ];
+function findLongestWord (array) {
 
-// Calculating a Sum
+var longuestWord = ""
+for (var i=0; i<array.length; i++) {
+if (array[i].length > longuestWord.length){
+  longuestWord= array[i]
+}
+}
+
+if (array.length === 0){
+  return
+}
+return longuestWord;
+
+
+}
+findLongestWord(words)
 
+// Calculating a Sum
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumArray (array){
+  var sum = 0;
+  for (i=0; i<array.length; i++){
+  sum = sum + array[i];
+  }
+  return sum
+}
+sumArray(numbers)
 
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers (array){
+var sumAverage = 0;
+for (i=0; i<array.length; i++){
+sumAverage = sumAverage + array[i];
+var average = sumAverage/array.length
+}
+return average
+}
```

Comment:
To keep your code dry (= avoid repetitions), you can also call the sumArray function that you created earlier.
``` javascript
function averageNumbers(arrayNumber) {
   var sumRes = sumArray(arrayNumber);
   if (arrayNumber.length > 0) {
   var average = sumRes / arrayNumber.length; 
   }
   return average;
 }
```

Reply:
the if statement here (  if (arrayNumber.length > 0) ) is to handle the case when we try to pass an empty array to the function, to please Jasmine :)